tests/e2e/matmul/generate_e2e_matmul_tests.py

```diff
@@ -420,14 +427,25 @@ class TestInputMatricesShapes:
 # Helper for generate_function. Generates TestInputMatricesShapes, i.e.
 # converts from the runtime shape dimensions in TestShape and given dynamicity to
 # the set of shapes to be used in a test function's input tensors.
-def generate_shapes(shape: TestShape, dynamicity: Dynamicity):
+def generate_shapes(shape: TestShape, transpose: TransposeCase, dynamicity: Dynamicity):
+    lhs_rows = shape_dim(shape.m, dynamicity)
+    lhs_cols = shape_dim(shape.k, dynamicity)
+    acc_rows = shape_dim(shape.m, dynamicity)
+    acc_cols = shape_dim(shape.n, dynamicity)
+    if transpose == TransposeCase.B:
+        rhs_rows = shape_dim(shape.n, dynamicity)
+        rhs_cols = shape_dim(shape.k, dynamicity)
+    else:
+        assert transpose == TransposeCase.NONE
```

Comment:
If you change the flag to a boolean as suggested above, then the `assert` wont be needed anymore and with the terminology aligned on "RHS" this will feel more natural to read.